examples/shapes_and_collections/hatch_demo.py

```diff
@@ -1,38 +1,69 @@
 """
-==========
-Hatch Demo
-==========
+=======
+Hatches
+=======
+
+Hatching (pattern filled polygons) is currently supported in the backends
+PS, PDF, SVG and Agg. The backends OSX, WX and Cairo ignore hatching.
+
+See also :doc:`/gallery/images_contours_and_fields/contourf_hatching` for
+an example using `~.Axes.contourf`.
 
-Hatching (pattern filled polygons) is supported currently in the PS,
-PDF, SVG and Agg backends only.
 """
+
 import numpy as np
 import matplotlib.pyplot as plt
-from matplotlib.patches import Ellipse, Polygon
+from matplotlib.patches import Ellipse, Polygon, Rectangle
 
 x = np.arange(1, 5)
 y1 = np.arange(1, 5)
 y2 = np.ones(y1.shape) * 4
 
-fig, (ax1, ax2, ax3) = plt.subplots(3)
+fig = plt.figure()
+axs = fig.subplot_mosaic([['bar1', 'patches'],['bar2', 'patches']])
 
-ax1.bar(x, y1, edgecolor='black', hatch="/")
-ax1.bar(x, y2, bottom=y1, edgecolor='black', hatch='//')
-ax1.set_xticks([1.5, 2.5, 3.5, 4.5])
+ax = axs['bar1']
+ax.bar(x, y1, edgecolor='black', hatch="/")
+ax.bar(x, y2, bottom=y1, edgecolor='black', hatch='//')
 
-ax2.bar(x, y1, edgecolor='black', hatch=['-', '+', 'x', '\\'])
-ax2.bar(x, y2, bottom=y1, edgecolor='black', hatch=['*', 'o', 'O', '.'])
-ax2.set_xticks([1.5, 2.5, 3.5, 4.5])
+ax = axs['bar2']
+ax.bar(x, y1, edgecolor='black', hatch=['-', '+', 'x', '\\'])
+ax.bar(x, y2, bottom=y1, edgecolor='black', hatch=['*', 'o', 'O', '.'])
 
-ax3.fill([1, 3, 3, 1], [1, 1, 2, 2], fill=False, hatch='\\')
-ax3.add_patch(Ellipse((4, 1.5), 4, 0.5, fill=False, hatch='*'))
-ax3.add_patch(Polygon([[0, 0], [4, 1.1], [6, 2.5], [2, 1.4]], closed=True,
-                      fill=False, hatch='/'))
-ax3.set_xlim((0, 6))
-ax3.set_ylim((0, 2.5))
+ax = axs['patches']
+ax.add_patch(Ellipse((4, 50), 10, 10, fill=True, hatch='*', facecolor='y'))
+ax.add_patch(Polygon([(10, 20), (30, 50), (50, 10)], hatch='\\/...',
+                     facecolor='g'))
+x = np.arange(0, 40, 0.2)
+ax.fill_between(x, np.sin(x) * 4 + 30, y2=0, hatch='///', zorder=2, fc='c')
+ax.set_xlim([0, 40])
+ax.set_ylim([10, 60])
+ax.set_aspect(1)
+plt.show()
 
+##############################################################################
+# Hatch style reference
+# ---------------------
+#
+# For convenience we document the 9 different possible hatch styles using the
+# code below.  Note that repeating the hatch character increases the density of
+# of the hatches, as shown in the first row and with "++".  Note also that
+# hatches can be combined ("/+O" in the first row).
+
+fig, axs = plt.subplots(3, 4, constrained_layout=True, figsize=(5, 4))
+
+hatches = ['/', '///', '/+O', '\\',
```

Comment:
Not a fan of this. This is a wild mixture of the basic styles and combinations, wich makes it hard to follow.

IMHO we should structure it

- first show the basic building blocks
- explain repetitions
- explain combinations 

Note also that `'\\'` is tricky because it's an escaped backslash not a repeated backslash.